### DISCUSSION TITLE
Error Connecting MongoDb Atlas with Prisma

### DISCUSSION THREAD

**Discussion by krishnaacharyaa** [prisma/prisma#19618] [Q&A]
### Bug description

When I connect my Nextjs app with prisma with my mongodb atlas I get the error saying:

```
An error occurred during DNS resolution: request timed out
   0: migration_core::commands::schema_push::Calculate `from`
             at migration-engine\core\src\commands\schema_push.rs:46
   1: migration_core::state::SchemaPush
             at migration-engine\core\src\state.rs:433
```

My url 
```
DATABASE_URL="mongodb+srv://Krish:password@cluster.jwxvr.mongodb.net/test"
```

schema.prisma
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  // url = "mongodb+srv://Krish:password@cluster0.n19uwxm.mongodb.net/test"
  url      = env("DATABASE_URL")
}

```

### How to reproduce

In prisma.schema
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}
```
My `DATABASE_URL` is `mongodb+srv://Krish:7qj-A9Qr.cJ-STK@cluster.jwxvr.mongodb.net/test`

Which is got by official mongodb
![image](https://github.com/prisma/prisma/assets/116620586/d90a0213-53ca-4856-9595-a03c7c3334b1)

### Prisma information

```prisma

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

```

### Environment & setup

- OS: Windows
- Database: MongoDb
- Node.js version: 19.6.0

### Prisma Version

```
4.11.0
```

**Comment by Frankcaozas** [comment]
same problem

**Reply by ludralph** [reply]
Hi @krishnaacharyaa 👋 

This is most likely an issue with DNS resolution. This means that your app is having trouble resolving the DNS address for your MongoDB Atlas cluster. You might need to make sure that your firewall or network settings are not blocking the connection to MongoDB Atlas.

**Reply by krishnaacharyaa** [reply]
Hi @ludralph ,
Other application with nodejs , having same connection from mongodb is able to access,
Only the prisma is giving problem.

I figured this is major issue and has many unanswered questions in SO aswell.
https://stackoverflow.com/questions/76200787/invalid-prisma-user-create-invocation-error-in-connectoran-error-occurred
https://stackoverflow.com/questions/76080476/getting-mongodb-error-an-error-occurred-during-dns-resolution-proto-error-unr
https://stackoverflow.com/questions/75457633/how-to-solve-prisma-connection-error-to-mongodb-atlas

Can you please look into the same. And possibly direct me where to look out for

**Reply by ludralph** [reply]
Hi @krishnaacharyaa, have you taken a look at the comment in this [issue](https://github.com/prisma/prisma/discussions/14908#discussioncomment-6077368). You can also try to run `nslookup <your_cluster_address>` in a terminal and see if it successfully resolves to an IP address.

**Reply by krishnaacharyaa** [reply]
@ludralph the issue which you referred has no solution

**Reply by krishnaacharyaa** [reply]
Though nslookup told me it couldn't find that domain , and it failed to lookup.

But when i run nodejs, connection is succesfully extablished !!!
```
const DB =
	"mongodb+srv://Krish:7qj-A9Qr.cJ-STK@cluster.jwxvr.mongodb.net/sample?retryWrites=true&w=majority";
mongoose
	.connect(DB)
	.then(() => {
		console.log("connection is successfully setup..");
	})
```
It logs `Connection is succesfully setup`

But the exact DB address when I use in prism it is not working, what might be the problem ??

**Comment by javierhbr** [comment]
I do have the same problem `DNS resolution` issue, on Windows 11. From the same laptop, I'm able to connect to the Mongo Atlas cluster using MongoCompass, but when I have to contest throw the NodeJS/Prisma Code I get the error. BTW on Mac works great.

![Screenshot 2023-12-13 at 2 46 12 PM](https://github.com/prisma/prisma/assets/2991184/0e23151a-615b-47b3-a172-4829b1081ca0)

UPDATE:

I fixed On windows changed the connection string from `mongodb+srv` to use the `mongodb:// {all the clusters hosts}:port` that works for me. For Mac and linux use the normal `mongodb+srv` format

Something like this:

```
mongodb://<username>:<password>@host1.mongodb.net:27017,host3.mongodb.net:27017,host3.mongodb.net:27017/?ssl=true&replicaSet=host1.shard-0&authSource=admin&retryWrites=true&w=majority";

**Reply by rafaelvcb** [reply]
I'm having the same issue, any fixes?

**Reply by javierhbr** [reply]
Hi @oxbar actually yes. On windows  I changed the connection string from `mongodb+srv` to use the `mongodb:// {all the clusters hosts}:port` that works for me. For Mac and linux use the normal `mongodb+srv` format

Something like this:

```
mongodb://<username>:<password>@host1.mongodb.net:27017,host3.mongodb.net:27017,host3.mongodb.net:27017/?ssl=true&replicaSet=host1.shard-0&authSource=admin&retryWrites=true&w=majority";
```

**Reply by rafaelvcb** [reply]
> Hi @oxbar actually yes. On windows I changed the connection string from `mongodb+srv` to use the `mongodb:// {all the clusters hosts}:port` that works for me. For Mac and linux use the normal `mongodb+srv` format
> 
> Something like this:
> 
> ```
> mongodb://<username>:<password>@host1.mongodb.net:27017,host3.mongodb.net:27017,host3.mongodb.net:27017/?ssl=true&replicaSet=host1.shard-0&authSource=admin&retryWrites=true&w=majority";
> ```

That worked for me, thanks!

**Comment by Lume98** [comment]
Upgrade to the latest version and you can use it now

I'm not sure which version started supporting it,curr version 5.17.0

mongodb+srv://username:password@cluster.xxxx.mongodb.net/xxxxx

**Reply by remia-dev** [reply]
can confirm this works

**Comment by Tiffany270** [comment]
perhaps relate to the OS, do not use `+srv` but `mongodb://<username>:<password>@<host1>:<port1>,<host2>:<port2>,<host3>:<port3>/<database>?replicaSet=<replicaSetName>&authSource=admin` ( u can find them in Connection info )

**Comment by toverux** [comment]
I had the same problem with Atlas+AWS, the strange thing is that I had two Ubuntu boxes, same version, same codebase, same network, and one didn't accept the `mongodb+srv://` scheme (ended up in immediate DNS resolution failure). So it's not a problem just on Windows.
The `dig` output for the domain was the same regarding DNS resolution (it's a bit magical, it uses an SOA to transfer control to AWS in my case, which I suspect is part of the problem?).

To find your `mongodb://` connection string, select a very old driver version (Mongo 2.x) which didn't support `+srv`, and you'll get a `mongodb://` connection string.

![image](https://github.com/user-attachments/assets/d02a5b0e-14b5-42c0-944d-be2a11f7da54)

**Reply by LuckyChimp** [reply]
That worked for me. Thank you very much!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
